src/xmppjs/HistoryManager.ts

```diff
@@ -0,0 +1,102 @@
+import { Element } from "@xmpp/xml";
+import { JID } from "@xmpp/jid";
+
+export interface IHistoryLimits {
+    maxChars?: number | undefined,
+    maxStanzas?: number | undefined,
+    seconds?: number | undefined,
+    since?: Date | undefined,
```

Comment:
The `?` and `| undefined` are redundant. I suggest you pick one, ideally consistent with existing code.